services/management/checks.go

```diff
@@ -212,24 +212,62 @@ func (s *ChecksAPIService) ListSecurityChecks(ctx context.Context, req *manageme
 		m[c] = struct{}{}
 	}
 
-	checks, err := s.checksService.GetChecks()
+	results, err := s.checksService.GetChecks()
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get available checks list")
 	}
 
-	res := make([]*managementpb.SecurityCheck, 0, len(checks))
-	for _, c := range checks {
-		_, disabled := m[c.Name]
-		res = append(res, &managementpb.SecurityCheck{
-			Name:        c.Name,
-			Disabled:    disabled,
-			Summary:     c.Summary,
-			Description: c.Description,
-			Interval:    convertInterval(c.Interval),
-		})
+	checks := make([]*managementpb.SecurityCheck, 0, len(results))
+	if req != nil && req.FilterParams != nil {
+		if req.FilterParams.Category != nil {
+			// we only support filtering by category for now.
+			// later, other "keys" can be supported by "AND-ing" the filters i.e., the response from each filter
+			// can be passed to the next.
+			filterValues := req.FilterParams.Category.Value
+			switch filterValues.(type) {
+			case *managementpb.InFilter_StringValues:
+				checks = filterByCategory(results, m, filterValues.(*managementpb.InFilter_StringValues).StringValues.Values)
+			default:
+				return nil, errors.New("no valid 'string_values' for category filter")
+			}
+		}
+	} else {
+		for _, c := range results {
+			_, disabled := m[c.Name]
+			checks = append(checks, &managementpb.SecurityCheck{
+				Name:        c.Name,
+				Disabled:    disabled,
+				Summary:     c.Summary,
+				Description: c.Description,
+				Interval:    convertInterval(c.Interval),
+				Category:    c.Category,
+			})
+		}
 	}
 
-	return &managementpb.ListSecurityChecksResponse{Checks: res}, nil
+	return &managementpb.ListSecurityChecksResponse{Checks: checks}, nil
+}
+
+func filterByCategory(checks map[string]check.Check, disabledChecks map[string]struct{}, categories []string) []*managementpb.SecurityCheck {
+	res := make([]*managementpb.SecurityCheck, 0, len(checks))
+
+	var disabled bool
+	for _, category := range categories {
```

Comment:
Yeah, just that I'm not sure we can gain much from that. Because we will end up processing all the checks (in all categories) only to later select the few categories we need. 

I think we cut that down with the current approach, as it ignores checks that don't match with the if check.

Reply:
Agree with @PavelKhripkov,
He means to do the same that was done for disabledChecks for categories: create map where key is category and value is empty struct. So you will be able just check if category of check exist in the map. And in case of more filters it will be simpler to add them too.

Reply:
Oh, got it.